modules/yahoosspBidAdapter.js

```diff
@@ -101,6 +101,37 @@ function extractUserSyncUrls(syncOptions, pixels) {
   return userSyncObjects;
 }
 
+/**
+ * @param {string} url
+ * @param {object} consentData
+ * @param {object} consentData.gpp
+ * @param {string} consentData.gpp.gppConsent
+ * @param {array} consentData.gpp.applicableSections
+ * @param {object} consentData.gdpr
+ * @param {object} consentData.gdpr.consentString
+ * @param {object} consentData.gdpr.gdprApplies
+ * @param {string} consentData.uspConsent
+ */
+function updateConsentQueryParams(url, consentData) {
+  const parameterMap = {
+    'gdpr_consent': consentData.gdpr.consentString,
+    'gdpr': consentData.gdpr.gdprApplies ? '1' : '0',
+    'us_privacy': consentData.uspConsent,
+    'gpp': consentData.gpp.gppString,
+    'gpp_sid': consentData.gpp.applicableSections ? consentData.gpp.applicableSections.join(',') : ''
+  }
```

Comment:
Also here as well I think you need to add safe checks because thee gdpr, gpp and uspConent objects might be undefined or null

```js
  const parameterMap = {
    'gdpr_consent': consentData.gdpr?.consentString,
    'gdpr': consentData.gdpr?.gdprApplies ? '1' : '0',
    'us_privacy': consentData?.uspConsent,
    'gpp': consentData.gpp?.gppString,
    'gpp_sid': consentData.gpp?.applicableSections ? consentData.gpp.applicableSections.join(',') : ''
  }
```

Something like that, (not tested)